src/lib/utils.ts

```diff
@@ -3,11 +3,26 @@ import { Course } from "@prisma/client";
 import { isAlpha } from "class-validator";
 import { v4 as uuidv4 } from "uuid";
 
+/**
+ * Zips two arrays together
+ * @param arr1 First array
+ * @param arr2 Second array
+ * @returns zip of two arrays
+ */
 export const zip = (arr1: any[], arr2: any[]) => {
   return arr1.map((elem, index) => [elem, arr2[index]]);
 };
 
 // Utility function to truncate tab title
+
+/**
+ * Truncates a title to a certain length
+ *
+ * NOTE: There is also a tailwindCSS class for this (truncate)
```

Comment:
Awesome. Let's refactor the old code that uses this function where the tailwind utility class can be used.

Reply:
My bad, the way you are using it is different than the tailwindcss class I should have taken a closer look.

[CSS Truncate](https://tailwindcss.com/docs/text-overflow) just adds ellipses only in the UI, it does not mutate the actual string. I see that your mutating the string and saving it with ellipses which is different behavior.

I am a bit reluctant to chase down all the providers / hooks to see if not mutating the string throws any errors (I can already seen in one case that `truncteTitle` enforces a description max length).